umh-core/pkg/service/redpanda_monitor/redpanda_monitor.go

```diff
@@ -0,0 +1,618 @@
+	s6service "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/s6"
+	"go.uber.org/zap"
+)
+
+// Ensure RedpandaMonitorService implements IRedpandaMonitorService
+var _ IRedpandaMonitorService = (*RedpandaMonitorService)(nil)
+
+type RedpandaMonitorService struct {
+	fs              filesystem.Service
+	logger          *zap.Logger
+	metricsState    *RedpandaMetricsState
+	s6Manager       *s6fsm.S6Manager
+	s6Service       s6service.Service
+	s6ServiceConfig *config.S6FSMConfig // There can only be one instance of this service (as there is also only one redpanda instance)
+}
+
+func NewRedpandaMonitorService(fs filesystem.Service) *RedpandaMonitorService {
+	log := logger.New(logger.ComponentRedpandaMonitorService, logger.FormatJSON)
+	return &RedpandaMonitorService{
+		fs:           fs,
+		logger:       log,
+		metricsState: NewRedpandaMetricsState(),
+		s6Manager:    s6fsm.NewS6Manager(logger.ComponentRedpandaMonitorService),
+		s6Service:    s6service.NewDefaultService(),
+	}
+}
+
+const START_MARKER = "BEGINBEGINBEGINBEGINBEGINBEGINBEGINBEGINBEGINBEGIN"
+const MID_MARKER = "MIDMIDMIDMIDMIDMIDMIDMIDMIDMIDMIDMIDMIDMIDMIDMIDMID"
+const END_MARKER = "ENDENDENDENDENDENDENDENDENDENDENDENDENDENDENDENDEND"
+
+func (s *RedpandaMonitorService) generateRedpandaScript() (string, error) {
+	// Build the redpanda command - curl http://localhost:9644/public_metrics
+	// Create the script content with a loop that executes redpanda every second
+	// Also let's use gzip to compress the output & hex encode it
+	// We use gzip here, to prevent the output from being rotated halfway through the logs & hex encode it to avoid issues with special characters
+	// Max-time: https://everything.curl.dev/usingcurl/timeouts.html
+	scriptContent := fmt.Sprintf(`#!/bin/sh
+while true; do
+  echo "%s"
+  curl -sSL --max-time 1 http://localhost:9644/public_metrics | gzip -c | xxd -p
+  echo "%s"
+  curl -sSL --max-time 1 http://localhost:9644/v1/cluster_config | gzip -c | xxd -p
+  echo "%s"
+  sleep 1
+done
+`, START_MARKER, MID_MARKER, END_MARKER)
+
+	return scriptContent, nil
+}
+
+func (s *RedpandaMonitorService) getS6ServiceName() string {
+	return "redpanda-monitor"
+}
+
+func (s *RedpandaMonitorService) GenerateS6ConfigForRedpandaMonitor() (s6serviceconfig.S6ServiceConfig, error) {
+	scriptContent, err := s.generateRedpandaScript()
+	if err != nil {
+		return s6serviceconfig.S6ServiceConfig{}, err
+	}
+
+	s6Config := s6serviceconfig.S6ServiceConfig{
+		Command: []string{
+			"/bin/sh",
+			fmt.Sprintf("%s/%s/config/run_redpanda_monitor.sh", constants.S6BaseDir, s.getS6ServiceName()),
+		},
+		Env: map[string]string{},
+		ConfigFiles: map[string]string{
+			"run_redpanda_monitor.sh": scriptContent,
+		},
+	}
+
+	return s6Config, nil
+}
+
+// GetConfig is not implemented, as the config is static
+
+// parseRedpandaLogs parses the logs of a redpanda service and extracts metrics
```

Comment:
you need to check the case where there was a logrotation done by S6. then the markers could be in different files.

for simplicity, you could simply ignore it in this case, so throw it to the fsm and let it automatically reconcile a couple of ticks later again.

Reply:
we already do this, if the markers are not found, we return an error.